src/session_base.cpp

```diff
@@ -536,6 +536,11 @@ void zmq::session_base_t::reconnect ()
     //  Reconnect.
     if (options.reconnect_ivl != -1)
         start_connecting (true);
+    else {
+        std::string *ep = new (std::string);
```

Comment:
when/how will this string be deleted?

Reply:
By the application thread on receipt of the message

Reply:
The pointer is stored in the union args_t which is part of struct command_t, and the receiver deletes it here:

https://github.com/zeromq/libzmq/pull/2756/files#diff-4243b1744be5585239d76c7e7d18017fR1428